components/commons/SearchInput.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import Image from "next/image";
+import styled from "styled-components";
+
+function SearchInput({
+  searchLink,
+  searchResult,
+  setSearchResult,
+}: {
+  searchLink: (keyword: string) => Promise<void>;
+  searchResult: string;
+  setSearchResult: React.Dispatch<React.SetStateAction<string>>;
+}) {
+  return (
+    <>
+      <SearchBox>
+        <input
+          onChange={(e) => {
+            searchLink(e.target.value);
+            setSearchResult(e.target.value);
+          }}
+          placeholder="링크를 검색해주세요"
+        />
```

Comment:
이 컴포넌트는 제어 컴포넌트기도 하고, 동작 상 value를 전달해야 안전하게 input value와 일치되게 동작할거같아요

이번 기회에 [이 부분 공식문서](https://react.dev/reference/react-dom/components/input)를 보고가시면 좋겟네요
```suggestion
        <input
          value={searchResult}
          onChange={(e) => {
            searchLink(e.target.value);
            setSearchResult(e.target.value);
          }}
          placeholder="링크를 검색해주세요"
        />
```